python_modules/dagster/dagster_tests/core_tests/asset_defs_tests/test_assets_job.py

```diff
@@ -1081,27 +1081,35 @@ def innermost_thing():
     def transformer(string):
         return string + "qux"
 
-    @graph(out={"n1": GraphOut(), "n2": GraphOut()})
+    @op
+    def combiner(s1, s2):
+        return s1 + s2
+
+    @graph(out={"n1": GraphOut(), "n2": GraphOut(), "unused": GraphOut()})
     def middle_thing():
         n1, unused_output = innermost_thing()
         n2 = get_string()
-        return {"n1": n1, "n2": n2}
+        return {"n1": n1, "n2": n2, "unused": unused_output}
 
-    @graph(out={"n1": GraphOut(), "n2": GraphOut()})
+    @graph(out={"n1": GraphOut(), "n2": GraphOut(), "unused": GraphOut()})
     def outer_thing(foo_asset):
-        n1, output = middle_thing()
+        n1, output, unused_output = middle_thing()
         n2 = transformer(output)
-        unused_output = transformer(foo_asset)  # pylint: disable=unused-variable
-        return {"n1": n1, "n2": n2}
+        unused_output = combiner(unused_output, transformer(foo_asset))
+        return {"n1": n1, "n2": n2, "unused": unused_output}
 
     @asset
     def foo_asset():
         return "foo"
 
-    thing_asset = AssetsDefinition(
+    thing_asset = AssetsDefinition.from_graph(
```

Comment:
the graph created in this test was invalid (because it had unused outputs), so this needed to updated.